### PR TITLE
feat: separate stoplight project component into dev-portal package

### PR DESCRIPTION
**Pull request by mpodlasin**
Relates to #1049 

The main goal of this PR is to keep the changes *as small as possible*, in order to avoid conflicts and have something that works ASAP.

Most notably this PR:
* moves only `StoplightProject` component to the new `elements-dev-portal` package. In future PRs I'll probably move a few more.
* does not include README
* might involve some duplication that *could* be potentially reduced (storybook config files come to mind here).
* does not move any styles (still requiring users to import styles from `elements` package)

### CODE REVIEW

**Review by philsturgeon — APPROVED** [review]

**Review by mmiask — COMMENTED** [review]
- [x] Elements build
- [x] Examples build
- [x] Examples run correctly (no crashing)
- [x] Demo builds and runs correctly

Other observations:
- `elements-dev-portal` Storybook opens to an error page
- `elements-dev-portal` web-components story does not run Docs
- Missing icons + styling issues (although as it is stated in this description, it's not in the scope of this PR)

Bear in mind that I did not yet go through the code, this is purely based on clicking around. I'll give more feedback once I read through it

File: examples/angular/angular.json
Comment (mallachari) [inline examples/angular/angular.json:54]: What's the reason for that change? Was it breaking earlier?

File: packages/elements-dev-portal/package.json
Comment (mallachari) [inline packages/elements-dev-portal/package.json:27]: There is a `build.styles` that runs `build-styles.js` during postbuild in elements package. Don't we need that here too?

File: packages/elements-dev-portal/web-components.config.js
Comment (mallachari) [inline packages/elements-dev-portal/web-components.config.js:25]:
```suggestion
        exclude: /node_modules/,
```

**Review by mallachari — COMMENTED** [review]
Looks good to me.
@mmiask, I can't see what you described. Storybook seems fine. I don't see the error page and web-components load correctly. Can you still reproduce that?

File: examples/angular/angular.json
Comment (mpodlasin) [reply examples/angular/angular.json:54]: We are now adding 2 web-components files - one for API and one for StoplightProject.

Shouldn't be an issue I think? I don't believe people will use *both* of those at the same time? :thinking:

Comment (mallachari) [reply examples/angular/angular.json:54]: Looks like it's about bundle size. I guess we exceeded limit with new package added.

File: packages/elements-dev-portal/package.json
Comment (mpodlasin) [reply packages/elements-dev-portal/package.json:27]: I haven't moved any styles yet, I plan to do it in a separate PR.

After this PR folks still need to import styles from `@stoplight/elements`.

File: packages/elements-dev-portal/tsconfig.build.json
Comment (mmiask) [inline packages/elements-dev-portal/tsconfig.build.json:16]: Same as the other comment - why do we need those 2 lines?

**Review by mmiask — COMMENTED** [review]
Looks good to me as well. I left a few questions, although they shouldn't block merging.

I'll approve once I confirm that Storybook indeed works as intended (re @mallachari comment)

**Comment by mmiask** [comment]
@mallachari Storybook behavior is not consistent for me. These are the 3 different ways it runs for me. No special reproduction steps, just checkout this branch and run `yarn elements-dev-portal storybook`

Screenshot 2021-04-28 at 14 10 19

---

Screenshot 2021-04-28 at 14 11 31

---

Screenshot 2021-04-28 at 14 12 00

**Comment by mpodlasin** [comment]
I am having the same issues as @mmiask 

Debugging currently.

File: tsconfig.json
Comment (mpodlasin) [reply tsconfig.json:9]: Yeah, tried it, and it crashes builds.

Presumably because `@stoplight/elements` actually doesn't fall under `@stoplight/elements/*` rule, because it doesn't end with `/`.

File: packages/elements-dev-portal/package.json
Comment (mpodlasin) [reply packages/elements-dev-portal/package.json:23]: @marcelltoth was right. We shouldn't put massive restrictions here, because for the most people just using elements, those restrictions are not necessary.

These are more restrictions of our internal tools.

**Review by mallachari — APPROVED** [review]
Works fine now

**Review by mmiask — APPROVED** [review]
Approving! It still runs a bit slow if you open the new endpoint (which might be more noticeable because of StoplightProject broken styling, it actually also happens in API component), but we can address that in the next PR which will include the styling

Great job with separating these two parts, this is already a big step in having the new structure 🎉

**Pull request merged**
